server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/migration/MappingsSourcesMigration.java

```diff
@@ -35,19 +37,23 @@
 
 public class MappingsSourcesMigration implements Migration {
     private static final Logger LOGGER = LoggerFactory.getLogger(MappingsSourcesMigration.class);
+
     private final CassandraRecipientRewriteTableDAO cassandraRecipientRewriteTableDAO;
     private final CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO;
+    private final AtomicLong processedMappingsCount;
 
     @Inject
     public MappingsSourcesMigration(CassandraRecipientRewriteTableDAO cassandraRecipientRewriteTableDAO,
                                     CassandraMappingsSourcesDAO cassandraMappingsSourcesDAO) {
         this.cassandraRecipientRewriteTableDAO = cassandraRecipientRewriteTableDAO;
         this.cassandraMappingsSourcesDAO = cassandraMappingsSourcesDAO;
+        this.processedMappingsCount = new AtomicLong(0);
     }
 
     @Override
     public Result run() {
         return cassandraRecipientRewriteTableDAO.getAllMappings()
+            .doOnNext(pair -> processedMappingsCount.incrementAndGet())
```

Comment:
you increase the count before mapping is executed. I prefer to move this statement to right after `.flatMap(this::migrate)`